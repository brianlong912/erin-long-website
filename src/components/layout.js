/**
 * Layout component that queries for data
 * with Gatsby's useStaticQuery component
 *
 * See: https://www.gatsbyjs.com/docs/use-static-query/
 */

import React, { useEffect } from "react"
import PropTypes from "prop-types"
import { useStaticQuery, graphql } from "gatsby"
import { FaArrowUp as ArrowUp } from "react-icons/fa"

import Header from "./header"
import "../styles/layout.css"

const Layout = ({ children }) => {
  const data = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `)

  useEffect(() => {
    window.addEventListener("scroll", handleScroll)
  })

  return (
    <>
      <Header siteTitle={data.site.siteMetadata?.title || `Title`} />
      <div style={{}}>
        <main>{children}</main>
        <ScrollTopButton />
        {/* <footer
          style={{
            marginTop: `2rem`,
          }}
        ></footer> */}
      </div>
    </>
  )
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
}

function ScrollTopButton() {
  var yPos;
  useEffect(() => {
    yPos = window.pageYOffset
  })
  function toTopFunction() {
    document.body.scrollTop = 0 //For Safari?
    document.documentElement.scrollTop = 0 //for other browsers
  }

  return (
    <button
      id="toTopButton"
      className="toTopButton"
      onClick={() => toTopFunction()}
      style={{
        opacity: yPos ? "1" : "0",
        visibility: yPos ? "visible" : "hidden",
        cursor: "pointer",
      }}
    >
      <ArrowUp style={{ width: "50px", height: "50px" }} />
    </button>
  )
}

function handleScroll() {
  var yPos = window.pageYOffset
  var toTopButton = document.getElementById("toTopButton")
  if (yPos) {
    toTopButton.style.opacity = "1"
    toTopButton.style.visibility = "visible"
  } else {
    toTopButton.style.opacity = "0"
    toTopButton.style.visibility = "hidden"
  }
}

export default Layout
