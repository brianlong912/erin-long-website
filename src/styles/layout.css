.toTopButton {
  position: fixed;
  display: flex;
  align-items: center;
  justify-content: center;
  bottom: 30px;
  right: 40px;
  z-index: 98;
  border-radius: 50%;
  height: 70px;
  width: 70px;
  border: solid 4px;
  background-color: rgba(70,70,70,0.6);
  color: white;
  transition: opacity 400ms;
}
.toTopButton:hover {
  color: #cfcfcf
}
.toTopButton:focus {
  outline: none;
}
