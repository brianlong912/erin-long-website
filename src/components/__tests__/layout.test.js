import React from "react"
import renderer from "react-test-renderer"
import { render, fireEvent, cleanup } from "@testing-library/react"

import { PureLayout as Layout } from "../layout"

afterEach(cleanup);

describe("Layout", () => {
  const data = {
    site: {
      siteMetadata: {
        title: "Erin Long",
      },
    },
  }
  it("renders correctly", () => {
    const tree = renderer.create(<Layout data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("shows toTopButton when scrolled down", () =>{
    const {rerender, container} = render(<Layout data={data} />)

    expect(container.querySelector("#toTopButton")).toHaveStyle("visibility: hidden")

    fireEvent.scroll(window, {target: {scrollY: 10}})

    rerender(<Layout data={data} />)

    expect(container.querySelector("#toTopButton")).toHaveStyle("visibility: visible")
  })
})
