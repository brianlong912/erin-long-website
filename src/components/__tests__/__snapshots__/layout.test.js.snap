// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
Array [
  <header
    style={
      Object {
        "background": "#191919",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "margin": "0 auto",
          "padding": "1.2rem 1.1rem",
        }
      }
    >
      <div
        style={
          Object {
            "width": "50%",
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "fontSize": "2.5rem",
              "textDecoration": "none",
            }
          }
        >
          Erin Long
        </a>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "height": "1.5em",
            "justifyContent": "flex-end",
            "width": "50%",
          }
        }
      >
        <a
          className="pageLink"
          href="/"
        >
          Photos
        </a>
        <a
          className="pageLink"
          href="/about"
        >
          About
        </a>
        <a
          className="pageLink"
          href="/contact"
        >
          Contact
        </a>
      </div>
    </div>
  </header>,
  <div
    style={Object {}}
  >
    <main />
    <button
      className="toTopButton"
      id="toTopButton"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "opacity": "0",
          "visibility": "hidden",
        }
      }
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
            "height": "50px",
            "width": "50px",
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
        />
      </svg>
    </button>
  </div>,
]
`;
